torch/sparse/_triton_ops.py

```diff
@@ -703,6 +703,102 @@ def bsr_dense_mm(
         _run_dense_rowspace_kernel(blocksize, values, crow_indices, col_indices, dense, out, max_grid)
 
         return out_backup
+
+
+    @triton.jit
+    def _bsr_softmax_kernel(
+        crow_indices_ptr,
+        crow_indices_batch_stride,
+        crow_indices_stride,
+        values_ptr,
+        values_batch_stride,
+        values_row_block_stride,
+        values_nnz_col_block_stride,
+        row_block, col_block,
+        MAX_ROW_NNZ: tl.constexpr
+    ):
+        batch_pid = tl.program_id(axis=2)
+        row_block_offset_pid = tl.program_id(axis=1)
+        row_block_pid = tl.program_id(axis=0)
+
+        crow_indices_offset_ptr = (
+            crow_indices_ptr
+            + crow_indices_batch_stride * batch_pid
+            + crow_indices_stride * row_block_pid
+        )
+        nnz_offset = tl.load(crow_indices_offset_ptr)
+        nnz_offset_next = tl.load(crow_indices_offset_ptr + crow_indices_stride)
+
+        # Compute nnz for the row with number row_block_pid.
+        # If it is zero, skip the row.
+        row_nnz = nnz_offset_next - nnz_offset
+        if row_nnz == 0:
+            return
+
+        row_arange = tl.arange(0, MAX_ROW_NNZ)
```

Comment:
Only a single row is loaded, not a block row, so the kernel becomes almost identical to the softmax example in the Triton tutorial.

Reply:
All right, so there is a cap in this kernel:
```python
In [1]: import torch                                                                                                                                                                                                                          
                                                                                                                                                                                                                                              
In [2]: x = torch.rand(1, 150000, device='cuda', dtype=torch.half)                                                                                                                                                                            
                                                                                                                       
In [3]: from torch.sparse._triton_ops import bsr_softmax                                                                                                                                                                                      
                                                           
In [4]: res1 = bsr_softmax(x.to_sparse_bsr(x.shape))                                                                   
<ipython-input-4-33c261e075e6>:1: UserWarning: Sparse BSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered 
internally at /home/nik/git/Quansight/pytorch/aten/src/ATen/SparseCsrTensorImpl.cpp:53.)                                                                                                                                                      
  res1 = bsr_softmax(x.to_sparse_bsr(x.shape)) 
...
CompilationError: at 30:30:        + crow_indices_stride * row_block_pid
    )
    nnz_offset = tl.load(crow_indices_offset_ptr)
    nnz_offset_next = tl.load(crow_indices_offset_ptr + crow_indices_stride)

    # Compute nnz for the row with number row_block_pid.
    # If it is zero, skip the row.
    row_nnz = nnz_offset_next - nnz_offset
    if row_nnz == 0:
        return

    row_arange = tl.arange(0, MAX_ROW_NNZ)
                              ^
ValueError('numel (262144) exceeds triton maximum tensor numel (131072)')

```

Reply:
@nikitaved - If it's a straightforward modification, I think it's worth it. Otherwise we can do it as a follow up.

Overall it's unlikely we'll see inner dimensions of that size in the context of SDPA.